src/types.ts

```diff
@@ -0,0 +1,38 @@
+import { Block } from 'web3-eth'
+import { provider } from 'web3-providers'
+import { TransactionReceipt, Transaction } from 'web3-core'
+import Web3 from 'web3'
+
+declare global {
+    interface Window {
+        ethereum: provider & {
+            enable: () => Promise<string[]>
+            on?: () => EventListener
+        }
+        web3: Web3
+    }
+}
+
+interface Web3Api {
+    web3: Web3
+    getCurrentAccount: () => Promise<string>
+    getAccounts: () => Promise<string[]>
+    getBalance: (account?: string) => Promise<string>
+    getBlock: (blockNumberOrString: string | number) => Promise<Partial<Block>>
+    getNetwork: () => Promise<number>
+    getNetworkName: (id?: number | string) => Promise<string>
+    getTransaction: (txHash: string) => Promise<Transaction>
+    getTransactionReceipt: (txHash: string) => Promise<TransactionReceipt>
+    // Utils
+    isReadOnly: () => boolean
```

Comment:
I don't think we need this. I would think this only make sense if we allow users to "sign in" by entering an address, so we show a read only version.

I think what you were using this for is to display a limited UI when there's an error connecting the wallet. But I think we can take other approaches (there's other open discussions about it, so no need to add it here too )